superset_patchup/views.py

```diff
@@ -0,0 +1,61 @@
+"""
+Additional extensions to SuperSet views
+"""
+
+import json
+from flask_appbuilder import expose
+from flask_appbuilder.models.sqla.interface import SQLAInterface
+import superset
+
+from . import (VERSION, __version__)
+
+class SupersetKetchupApiView(superset.views.base.BaseSupersetView):
+
+    """
+    Extensions to Superset API
+    """
+
+    def __init__(self):
+        self.route_base = "/superset-ketchup/api"
+
+    @superset.views.base.api
+    @expose('/version/', methods=['GET'])
+    def version(self):
+        """Returns *superset-api* version"""
+        return self.json_response({ 'version': VERSION, 'versionStr': __version__ })
+
+    @superset.views.base.api
+    @expose('/all_dashboards/', methods=['GET'])
+    def all_dashboards(self):
+        """
+        Extension to allow API access to information *about* dashboards.
+        See: /fave_dashboards/, etc.
+        Returns the ids, links, names, and URLs of the dashboards a user has access to.
+        """
+
+        dashboards = superset.db.session.query(
+            superset.models.core.Dashboard)
+
+        # Query only dashboards where we have at least some access - note that this exact query
+        # filter is used in the core Superset view to prune visible dashboards.
+        # Logic is basically "dashboard owned by user or at least one slice accessible by user"
+        # See: https://github.com/apache/incubator-superset/blob/0.27/superset/views/core.py#L154
+        dashboards = (superset.views.core.DashboardFilter('slice',
```

Comment:
Here's some magic - this is the filter used by superset to figure out which dashboards to "list" in the view.  Basically "if a user owns the db or has permission to one or more slices".  Happy to change around though if we decide we want something more particular.